pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -90,6 +90,20 @@ def __verify_train_loop_configuration(self, model: "pl.LightningModule") -> None
                 "(rather, they are called on every optimization step)."
             )
 
+    def check_progress_bar(self, model: "pl.LightningModule") -> None:
+        r"""
+        Checks if get_progress_bar_dict is overriden and sends a deprecation warning.
+
+        Args:
+            model: The model to check the get_progress_bar_dict method.
+
+        """
+        if is_overridden("get_progress_bar_dict", model):
+            rank_zero_deprecation(
+                "The `LightningModule.get_progress_bar_dict` method was deprecated in v1.5 and will be removed in v1.7."
+                "Please use the `trainer.progress_bar_callback.get_progress_bar_dict` instead."
+            )
+
```

Comment:
We could add this as a private method that runs at the end of `verify_loop_configurations`. It doesn't need to be its own top-level entity. We could leave a todo for ourselves to delete this in v1.7.